src/app/components/algorithms/algorithm-view/algorithm-view.component.ts

```diff
@@ -42,7 +44,9 @@ export class AlgorithmViewComponent implements OnInit, OnDestroy {
 
   links: BreadcrumbLink[] = [{ heading: '', subHeading: '' }];
 
-  private routeSub: Subscription;
+  private dataSubscription: Subscription;
```

Comment:
```suggestion
  private dataSubscription: Subscription|null = null;
```
This stricter typing will be required in later versions of angular/typescript anyway...